pkg/apis/crd/v1beta1/types.go

```diff
@@ -154,3 +156,109 @@ type ControllerCondition struct {
 	// Human readable message indicating details
 	Message string `json:"message,omitempty"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterGroup struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Desired state of the group.
+	Spec GroupSpec `json:"spec"`
+	// Most recently observed status of the group.
+	Status GroupStatus `json:"status"`
+}
+
+type GroupSpec struct {
+	// Select Pods matching the labels set in the PodSelector in
+	// AppliedTo/To/From fields. If set with NamespaceSelector, Pods are
+	// matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+	// Select all Pods from Namespaces matched by this selector, as
+	// workloads in AppliedTo/To/From fields. If set with PodSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except PodSelector.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	// IPBlocks describe the IPAddresses/IPBlocks that are matched in to/from.
+	// IPBlocks cannot be set as part of the AppliedTo field.
+	// Cannot be set with any other selector or ServiceReference.
+	// +optional
+	IPBlocks []v1alpha1.IPBlock `json:"ipBlocks,omitempty"`
```

Comment:
Should this `IPBlock` be upgraded to v1beta1 as well?

Reply:
Since this one is also used by NP, I upgrade it on PR #5186. Let me know your thoughts, something like we upgrade IPBlock in this PR or we squash these two PRs into one PR.

Reply:
I feel it's Ok to keep them in two PRs.